packages/hydrogen/src/product/VariantSelector.ts

```diff
@@ -32,19 +32,25 @@ type VariantSelectorProps = {
   variants?:
     | PartialDeep<ProductVariantConnection>
     | Array<PartialDeep<ProductVariant>>;
+  /** By default all products are under /products. Use this prop to provide a custom path. */
+  productPath?: string;
```

Comment:
Path options to me are always confusing because I don't know if I'm supposed to provide leading or ending `/` in it.
Can we either:
- Mention slashes are not required in the doc
- Trim slashes in JS
- Use a type that makes it clear ([ts playground](https://www.typescriptlang.org/play?ssl=1&ssc=114&pln=1&pc=1#code/C4TwDgpgBAcg9gdQE4EMxgJYDsDmBlAGxQGcALCYgHjyggA9gIsATYqY4JbHAPigF4oNeoxZsABgHoAJAG9sAMwhIoAJQC+4qAH4oWCADdlUAFxDaDJqyji5i4xslbd+oyrN4A3AChvCgK5YAMbAGHBYUABqKFwoWMB4EAQQIXBIlAAqFqLWHFy4PAAUcGDAZrJQYEhwzP4hAAoowKRm8MhomLiEJORUGXzqAJRQsuq+0bHxicmpSIWyVTV1wI3NZgDki7UhxOtDPhMYcQlJKcBp81vLqy1Qm9XbwMSSzBAQYHuDBzFHU6ezlwe1yat3WkiuO0+30mJxm5zmCyBDRBGwhT0kUKAA)):
  ```ts
  type NoWrappingSlashes<S extends string> =
    S extends `/${infer R}` ? never : S extends `${infer R}/` ? never : S;
  ```

Reply:
Good point. Probably best to make it work both with and without a leading slash, so it works well for non-TS folks too.